integration-tests/mockbean/src/test/java/io/quarkus/it/mockbean/GreetingResourceTest.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.it.mockbean;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.MockBean;
+
+@QuarkusTest
+class GreetingResourceTest {
+
+    @MockBean
+    MessageService messageService;
```

Comment:
Since these are not CDI injection points it could happen that ArC will consider the resolved beans removable. Maybe we should handle this use case and mark them unremovable?

Reply:
Like @stuartwdouglas said, if the bean has been removed, mocking won't change anything, so how would the user see the results of the their test?
I don't think this scenario is actually possible. 
It's likely I think that if normally the bean would be removed in the prod app, the user would be injecting some other bean into the test to check the effect of the mock, thus making the mock unremovable.